core/src/main/java/io/confluent/rest/SslFactoryPemHelper.java

```diff
@@ -0,0 +1,390 @@
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class provides util methods to handle Pem type keystore.
+ * Based on the provider type, an intermediate keystore of appropriate type will be
+ * created.
+ */
+public class SslFactoryPemHelper {
+
+  public static final String FIPS_KEYSTORE_TYPE = "BCFKS";
+  public static final String NONFIPS_KEYSTORE_TYPE = "PKCS12";
+  public static final String FIPS_SSL_PROVIDER = "BCJSSE";
+  public static final String FIPS_PROVIDER = "BCFIPS";
+  public static final String PEM_TYPE = "PEM";
+
+  private static final Logger log = LoggerFactory.getLogger(SslFactoryPemHelper.class);
+
+  public static boolean isPemSecurityStore(String keyStoreType) {
+    return Objects.equals(PEM_TYPE, keyStoreType);
+  }
+
+  public static KeyStore getKeyStoreFromPem(String keyStorePath, String keyStoreType,
+                                            Password keyPassword, String provider,
+                                            boolean isKeyStore) {
+    if (Objects.equals(FIPS_SSL_PROVIDER, provider)) {
+      if (!Objects.equals(PEM_TYPE, keyStoreType)) {
+        throw new RuntimeException(
+            String.format("Only %s security store supported with %s ssl provider",
+                PEM_TYPE, FIPS_SSL_PROVIDER));
+      }
+    }
+    if (keyPassword.value().equals("")) {
+      keyPassword = null;
+    }
+    FileBasedPemStore store = new FileBasedPemStore(
+        keyStorePath, keyPassword, isKeyStore, useBcfks(provider));
+    return store.get();
+  }
+
+  private static boolean useBcfks(String provider) {
+    return Objects.equals(FIPS_SSL_PROVIDER, provider);
+  }
+
+  public static String getKeyStoreType(String provider) {
+    return useBcfks(provider) ? FIPS_KEYSTORE_TYPE : NONFIPS_KEYSTORE_TYPE;
+  }
+
+  interface SecurityStore {
+    KeyStore get();
+
+    char[] keyPassword();
+
+    boolean modified();
+  }
+
+  // package access for testing
+  static class FileBasedStore implements SecurityStore {
+    private final String type;
+    protected final String path;
+    private final Password password;
+    protected final Password keyPassword;
+    private Long fileLastModifiedMs;
+    private KeyStore keyStore;
+    protected final boolean useBcfks;
+
+    FileBasedStore(String type, String path, Password password, Password keyPassword,
+                   boolean isKeyStore, boolean useBcfks) {
+      Objects.requireNonNull(type, "type must not be null");
+      this.type = type;
```

Comment:
Objects.requireNonNull(type, "type must not be null");
      this.type = type;

can be reduced to this.type = Objects.requireNonNull(type, "type must not be null");